test/src/e2e.dynval/dv.n+1.test.ts

```diff
@@ -0,0 +1,265 @@
+                    Error(
+                        `Expected chain_getPossibleAuthors to return an array of string or null but it returned ${result}`
+                    )
+                );
+            })
+            .catch(reject);
+    });
+}
+
+chai.use(chaiAsPromised);
+
+const [betty, , , ...otherDynValidators] = originalValidators;
+const allDynValidators = [betty, ...otherDynValidators];
+
+describe("Dynamic Validator N -> N+1", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const termSeconds = 20;
+
+    async function beforeInsertionCheck(checkingNode: CodeChain) {
+        const blockNumber = await checkingNode.sdk.rpc.chain.getBestBlockNumber();
+        const termMedata = await stake.getTermMetadata(
+            checkingNode.sdk,
+            blockNumber
+        );
+        const currentTermInitialBlockNumber =
+            termMedata!.lastTermFinishedBlockNumber + 1;
+        const validatorsBefore = await getPossibleAuthors(
+            checkingNode.sdk,
+            currentTermInitialBlockNumber
+        );
+
+        expect(termMedata!.currentTermId).to.be.equals(1);
+        expect(validatorsBefore!.length).to.be.equals(
+            otherDynValidators.length
+        );
+        expect(validatorsBefore).contains.all.members(
+            otherDynValidators.map(validator => validator.platformAddress.value)
+        );
+    }
+
+    async function bettyInsertionCheck(checkingNode: CodeChain) {
+        const margin = 1.2;
+        await checkingNode.waitForTermChange(2, termSeconds * margin);
+
+        const blockNumber = await checkingNode.sdk.rpc.chain.getBestBlockNumber();
+        const termMedata = await stake.getTermMetadata(
+            checkingNode.sdk,
+            blockNumber
+        );
+        const currentTermInitialBlockNumber =
+            termMedata!.lastTermFinishedBlockNumber + 1;
+        const validatorsAfter = await getPossibleAuthors(
+            checkingNode.sdk,
+            currentTermInitialBlockNumber
+        );
+
+        expect(termMedata!.currentTermId).to.be.equals(2);
+        expect(validatorsAfter).contains.all.members(
+            allDynValidators.map(validator => validator.platformAddress.value)
+        );
+    }
+
+    describe("Nominate a new candidate and delegate", async function() {
+        const nodes = withNodes(this, {
+            promiseExpect,
+            validators: otherDynValidators.map((signer, index) => ({
+                signer,
+                delegation: 5_000,
+                deposit: 10_000_000 - index // tie-breaker
+            })),
+            overrideParams: {
+                termSeconds,
+                maxNumOfValidators: 8
+            }
+        });
+
+        it("betty should be included in validators", async function() {
+            const checkingNode = nodes[0];
+            const faucetSeq = await checkingNode.sdk.rpc.chain.getSeq(
+                faucetAddress
+            );
```

Comment:
Let's call `beforeInsertionCheck` function here.

Reply:
Ok, I called it.